store/postgres/src/relational_queries.rs

```diff
@@ -1270,16 +1303,43 @@ impl SortKey {
     ///   order by [name direction,] id
     fn order_by(&self, out: &mut AstPass<Pg>) -> QueryResult<()> {
         out.push_sql("order by ");
-        if let Some(name) = &self.name {
-            out.push_identifier(name.as_str())?;
-            out.push_sql(" ");
-            out.push_sql(self.direction.to_sql());
-            out.push_sql(" nulls last");
-            if name.as_str() != PRIMARY_KEY_COLUMN {
-                out.push_sql(", ");
-                out.push_identifier(PRIMARY_KEY_COLUMN)?;
+        if let Some(column) = &self.column {
+            match &column.column_type {
+                ColumnType::TSVector((_, algorithm)) => {
+                    let sort_value = &self
+                        .value
+                        .as_ref()
+                        .and_then(|v| v.clone().as_string())
+                        .expect("fulltext search queries can only use EntityFilter::Equal");
```

Comment:
Generating errors during SQL generation (this function is ultimately called from `walk_ast` somewhere) leads to really confusing and hard to follow error messages because of how things bubble up in Diesel. This check should be done when the `SortKey` is constructed, and it should therefore have a `value: Option<String>` or (preferrably) `value: Option<&str>` field.